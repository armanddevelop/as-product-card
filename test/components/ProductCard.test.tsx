import React from 'react';
import render from "react-test-renderer";
//import {fireEvent} from "@testing-library/dom";
import {ProductCard} from '../../src/components/ProductCard';
import { product2 } from '../data/products';

const {act} = render;

describe("Product Card component UT",()=>{
  test("Test the product card component",()=>{
    const wrapper = render.create(
      <>
        <ProductCard product={product2}>
          {
            ()=>(
              <h1>Product card</h1>
            )
          }
        </ProductCard>
      </>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
   test("increaseBy UT",()=>{
    const mokedEvent:any={
         target:{
           name:"add-count",
           value:1
         }
    }
    const wrapper = render.create(
      <>
        <ProductCard product={product2}>
          {
            ({count,increaseBy})=>(
              <>
                <h1>This is the value of</h1>
                <span>{count}</span>
                <button value={1} name="add-count" onClick={(e)=>increaseBy(e)}></button>
              </>
            )
          }
        </ProductCard>
      </>
    );
    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
    act(()=>{
      (tree as any).children[2].props.onClick(mokedEvent);
    });
    tree =wrapper.toJSON();
    expect((tree as any).children[1].children[0]).toBe('1');
  });
});