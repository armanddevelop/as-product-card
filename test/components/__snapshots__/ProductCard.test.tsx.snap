// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Card component UT Test the product card component 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product card
  </h1>
</div>
`;

exports[`Product Card component UT increaseBy UT 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    This is the value of
  </h1>
  <span>
    0
  </span>
  <button
    name="add-count"
    onClick={[Function]}
    value={1}
  />
</div>
`;
